zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java

```diff
@@ -62,6 +65,38 @@ public CompletionStage<PendingIncidentUpdateBatch> getPendingIncidentsBatch(
         .thenApplyAsync(this::createPendingIncidentBatch, executor);
   }
 
+  @Override
+  public CompletionStage<Map<String, IncidentDocument>> getIncidentDocuments(
+      final List<String> incidentIds) {
+    final var request = createIncidentDocumentsRequest(incidentIds);
+
+    return client
+        .search(request, IncidentEntity.class)
+        .thenApplyAsync(this::createIncidentDocuments, executor);
+  }
+
+  private SearchRequest createIncidentDocumentsRequest(final List<String> incidentIds) {
+    final var idQ = QueryBuilders.ids(i -> i.values(incidentIds));
+    final var partitionQ =
```

Comment:
:thought_balloon: This may not be really necessary, because the partition ID is actually encoded in the key/ID - but this relies on the fact that ID == key, which I'm not super keen on. So I added the filter here.